xds/src/main/java/io/grpc/xds/internal/certprovider/CertificateProvider.java

```diff
@@ -42,6 +44,41 @@
     void onError(Status errorStatus);
   }
 
+  @VisibleForTesting
+  static final class DistributorWatcher implements Watcher {
+    @VisibleForTesting
+    final ArrayList<Watcher> downsstreamWatchers = new ArrayList<>();
```

Comment:
What's the reason of using ArrayList rather than HashSet? Could there be duplicate entries in the collection? Do the order of the entries matter?

Reply:
Order definitely doesn't matter. And I cannot think of any straightforward example that requires duplicate entries support. For now I will change it to HashSet.